### PR TITLE
Add tag master for version develop of dealii.

### PR DESCRIPTION
**Pull request by adamqc**
Add tag for version develop to allow caching of git repos.

### CODE REVIEW

File: var/spack/repos/builtin/packages/dealii/package.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ class Dealii(CMakePackage):
     version('8.3.0', 'fc6cdcb16309ef4bea338a4f014de6fa')
     version('8.2.1', '71c728dbec14f371297cd405776ccf08')
     version('8.1.0', 'aa8fadc2ce5eb674f44f997461bf668d')
-    version('develop', git='https://github.com/dealii/dealii.git')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/dealii/package.py]: why did you need this change? It should (and it does as far as I know) checkout `master` branch by default.

Comment (tgamblin) [inline var/spack/repos/builtin/packages/dealii/package.py]: @adamqc: mind answering this?

**Comment by adamqc** [comment]
Without the tag parameter, each time we build the package, the souce code need to be checkouted from the internet, i.e., the develop version won't get cached.
Sometimes we may want to build another branch but master, this can be achieved by create dealii-develop.tar.gz using git archive command and put it into the cache directory.

**Comment by davydden** [comment]
> Sometimes we may want to build another branch but master

out of curiosity, which other branches but `master` do you use for building? In `dealii` additional branches are only those for major.minor releases i.e. `dealii-8.4` and `dealii-8.3` etc. But the code from those branches is exactly what you get from bugfix releases like `8.4.1`, `8.4.2`, `8.2.1`, etc.

**Comment by adamqc** [comment]
When I say branch I mean personal branches or some specific commits. Before we submit a pull request, we want to test it with various specs.

**Comment by adamqc** [comment]
@davydden Can this PR be merged to `develop` now?
